water/lib/Dialect/Wave/IR/WaveAttrs.cpp

```diff
@@ -180,14 +244,31 @@ bool WaveHyperparameterAttr::hasSymbol(StringRef symbolName) const {
   return getMapping().get(symbolName) != nullptr;
 }
 
+//===----------------------------------------------------------------------===//
+// WaveSymbolAttr
+//===----------------------------------------------------------------------===//
+
+LogicalResult
+WaveSymbolAttr::verify(function_ref<InFlightDiagnostic()> emitError,
+                       StringRef name) {
+  static const llvm::StringSet<> indexSymbols(llvm::from_range,
+                                              WaveIndexSymbolAttr::AllCases);
+  if (indexSymbols.contains(name))
+    return emitError() << "illegal wave symbol name '" << name
+                       << "'. Did you mean to use WaveIndexSymbolAttr instead?";
```

Comment:
Nit: usually, we don't start new sentences in error messages. Consider adding a note to the diagnostic isntead.